src/platform/Zephyr/DiagnosticDataProviderImpl.cpp

```diff
@@ -212,7 +212,27 @@ CHIP_ERROR DiagnosticDataProviderImpl::GetNetworkInterfaces(NetworkInterface **
             ifp->hardwareAddress = ByteSpan(ifp->MacAddress, addressSize);
         }
 
-        head = ifp;
+        // Assuming IPv6-only support
+        Inet::InterfaceAddressIterator interfaceAddressIterator;
```

Comment:
Why in particular this Zephyr platform is requiring this loop in your update ?

Reply:
That is because `InterfaceIterator` doesn't provide access to the IPv4/IPv6 addresses information. In turn using only `InterfaceAddressIterator` would not allow to get information like interface type or hardware address.